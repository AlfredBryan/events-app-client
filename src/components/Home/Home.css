@import url("https://fonts.googleapis.com/css?family=Montserrat:400,700");
@import url("https://fonts.googleapis.com/css?family=Srisakdi:700");

.home-page {
  margin-top: 20px;
}

.link {
  cursor: pointer;
}
a {
  text-decoration: none;
  color: black;
}
a:hover {
  color: black;
}
.events {
  width: 295.01px;
  margin-top: 20px;
  height: 188px;
  transition: all 1s linear;
}

.events:hover {
  border: 0.02rem solid rgba(0, 0, 0, 0.2);
  box-shadow: 2px 2px 5px 0px rgba(0, 0, 0, 0.2);
}
.event-main {
  width: 947px;
  margin-left: 79px;
  margin-top: 20px;
}

.event-name {
  width: 149.5px;
  height: 61px;
  background: #6ec7e3;
  overflow: hidden;
}

.event-name h5 {
  text-transform: capitalize;
  font-family: "Montserrat", sans-serif;
  font-weight: bold;
  font-size: 18px;
  color: #000000;
  padding-top: 12px;
}
.description {
  width: 149.5px;
  height: 127px;
  background: #f2efec;
  overflow: hidden;
}

.description h5 {
  text-transform: capitalize;
  font-family: "Montserrat", sans-serif;
  font-weight: bold;
  font-size: 18px;
  color: #000000;
  padding-top: 12px;
}

.description a {
  font-size: 12px;
  color: red;
  font-family: "Srisakdi", cursive;
  text-decoration: none;
}

.home-footer {
  width: 947px;
  height: 61px;
  background: #1f1c1c;
  display: inline-block;
  bottom: 0px;
  margin-top: 100px;
}

@media (max-width: 500px) {
  .event-main {
    width: 400px;
    margin-left: 0px;
  }
  .home-footer {
    width: 400px;
  }
}
