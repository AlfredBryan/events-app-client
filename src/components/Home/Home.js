import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";

import NavBar from "../NavBar/NavBar";
import "./Home.css";
import Spinner from "../hoc/Spinner";

class Home extends Component {
  state = {
    events: [],
    loading: true
  };

  logOut = () => {
    localStorage.removeItem("token");
    this.props.history.replace("/login");
  };

  componentDidMount() {
    axios.get("https://events-apps.herokuapp.com/api/events/all").then(res => {
      const events = res.data.slice(0, 6);
      const updatedEvents = events.map(event => {
        return {
          ...event
        };
      });
      this.setState({
        events: updatedEvents,
        loading: false
      });
    });
  }

  render() {
    let { events, loading } = this.state;

    if (loading) {
      return <Spinner />;
    }

    if (events.length === 0) {
      return (
        <div>
          <NavBar />
          <div className="container wrapper home-page">
            <div className="row">
              <div className="col-9 mx-auto col-md-6 col-lg-4 link">
                <Link to="/create/event">
                  <i className="fa fa-plus" style={{ fontSize: "20px" }}>
                    <span style={{ padding: "10px" }}>Create Events</span>
                  </i>
                </Link>
              </div>

              <div className="col-9 mx-auto col-md-6 col-lg-4 link">
                <Link to="/users/view">
                  <i className="fa fa-users" style={{ fontSize: "20px" }}>
                    <span style={{ padding: "10px" }}>
                      View registered Users
                    </span>
                  </i>
                </Link>
              </div>
              <div
                className="col-9 mx-auto col-md-6 col-lg-4 link"
                onClick={this.logOut}
              >
                <i className="fa fa-power-off" style={{ fontSize: "20px" }}>
                  <span style={{ padding: "10px" }}>Log Out</span>
                </i>
              </div>
            </div>
            <div className="row">
              <div className="col-10 mx-auto text-center text-title text-uppercase pt-5">
                <h1 className="display-3">404</h1>
                <h2>No Events Yet</h2>
              </div>
            </div>
            <div className="home-footer"> </div>
          </div>
        </div>
      );
    } else {
    }
    return (
      <React.Fragment>
        <NavBar />
        <div className="container wrapper home-page">
          <div className="row">
            <div className="col-9 mx-auto col-md-6 col-lg-4 link">
              <Link to="/create/event">
                <i className="fa fa-plus" style={{ fontSize: "20px" }}>
                  <span style={{ padding: "10px" }}>Create Events</span>
                </i>
              </Link>
            </div>

            <div className="col-9 mx-auto col-md-6 col-lg-4 link">
              <Link to="/users/view">
                <i className="fa fa-users" style={{ fontSize: "20px" }}>
                  <span style={{ padding: "10px" }}>View registered Users</span>
                </i>
              </Link>
            </div>
            <div
              className="col-9 mx-auto col-md-6 col-lg-4 link"
              onClick={this.logOut}
            >
              <i className="fa fa-power-off" style={{ fontSize: "20px" }}>
                <span style={{ padding: "10px" }}>Log Out</span>
              </i>
            </div>
          </div>
          <div className="row event-main" id="event-main">
            {events.map(event => (
              <div className="col-xs-9 col-md-6 col-lg-4 events" key={event.id}>
                <div className="d-flex">
                  <Link to={`/signedup/${event.id}`}>
                    <div>
                      <img
                        src={event.image}
                        className="img-responsive"
                        style={{ width: "145.51px", height: "185px" }}
                        alt="event"
                      />
                    </div>
                  </Link>
                  <div>
                    <div className="event-name">
                      <h5>{event.name}</h5>
                    </div>
                    <div className="description">
                      <h5>{event.description}</h5>{" "}
                      <Link to={`/event/signup/${event.id}`}>
                        <p>Follow the link to Signup</p>
                      </Link>
                    </div>
                  </div>
                </div>
              </div>
            ))}
          </div>
        </div>
        <div id="home-footer" className="home-footer">
          {" "}
        </div>
      </React.Fragment>
    );
  }
}

export default Home;
