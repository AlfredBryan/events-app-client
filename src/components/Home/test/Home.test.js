import React from "react";
import { shallow } from "enzyme";
import Home from "../Home";

jest.mock("axios", () => {
  const testEvents = {
    events: []
  };

  return {
    get: jest.fn(() => Promise.resolve(testEvents))
  };
});

const axios = require("axios");

it("Fetch registered Users #componentDidMount", done => {
  const wrapper = shallow(<Home />);
  wrapper.instance().componentDidMount();
  expect(axios.get).toHaveBeenCalled();
  expect(axios.get).toHaveBeenCalledWith(
    "https://events-apps.herokuapp.com/api/events/all"
  );
  expect(wrapper.state()).toMatchSnapshot("events", []);
  done();
});
