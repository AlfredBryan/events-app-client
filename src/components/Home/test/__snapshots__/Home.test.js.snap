// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch registered Users #componentDidMount: events 1`] = `
Object {
  "events": Array [],
  "loading": true,
}
`;
